nf_core/components/install.py

```diff
@@ -50,6 +50,15 @@ def __init__(
             self.installed_by = [self.component_type]
 
     def install(self, component: str, silent: bool = False) -> bool:
+        if isinstance(component, dict):
+            if component.get("git_remote") is not None:
+                remote_url = component.get("git_remote")
+                branch = component.get("branch")
+                self.modules_repo = ModulesRepo(remote_url, branch)
+            else:
+                self.modules_repo = ModulesRepo(self.current_remote, self.branch)
+            component = component.get("name")
```

Comment:
There's always a "name" in the dictionary
```suggestion
            component = component["name"]
```